src/Components/InfiniteScroll.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import { IModel, IModelData, IQueryBuilder, IQueryManager, IQueryParams, QueryBuilder } from "redux-data-service";
+import { Omit } from "redux-data-service/dist/Omit";
+import { compose, defaultProps, lifecycle, pure, setDisplayName, withProps, withState, withStateHandlers } from "recompose";
+import { debounce } from "lodash";
+
+import { Query } from "../Query";
+import { withModelQuery } from "../WithModelQuery";
+
+export interface IInfiniteScrollProps<T extends IModelData> {
+  modelName: string;
+  query: IQueryParams | IQueryBuilder;
+  containerComponent: React.ComponentType<{ onScroll: React.UIEventHandler } & any>;
+  modelComponent: React.ComponentType<{ model: T } & any>;
+  containerProps?: any;
+  modelComponentProps?: any;
+  debounceTime?: number;
+}
+
+export interface IInfiniteScrollInternalProps<T extends IModel<IModelData>> extends Omit<IInfiniteScrollProps<any>, "query"> {
+  query: IQueryManager<T>;
+  updateQuery: (queryBuilder: IQueryBuilder) => void;
+  lastScrollTop: number;
+  handleScrollDebounced: (e) => void;
+  handleScrollPersistingEvent: (e) => void;
+}
+
+export interface IDisplayPreviousPageProps<T extends IModelData> {
+  modelName: string;
+  queryManager: IQueryManager<T>;
+  modelComponent: React.ComponentType<{ model: T } & any>;
+  modelComponentProps?: any;
+}
+
+const DisplayPreviousPage = compose<IDisplayPreviousPageProps<any>, IDisplayPreviousPageProps<any>>(
+  setDisplayName("DisplayPreviousPage"),
+  pure,
+)(
+  (({ queryManager, modelName, modelComponent: ModelComponent, modelComponentProps }) => {
+    return queryManager.hasPreviousPage() && (
+      <Query key={`page-${queryManager.response.previousPage}`} modelName={modelName} query={queryManager.getPreviousPage()}>
+        {({ query }) => (
+          <>
+            <DisplayPreviousPage queryManager={query} modelComponent={ModelComponent} modelComponentProps={modelComponentProps} modelName={modelName} />
+            {query.items.map(model => (
+              <ModelComponent key={model.id} model={model} {...modelComponentProps} />
+            ))}
+          </>
+        )}
+      </Query>
+    );
+  }),
+);
+
+export const InfiniteScroll = compose<IInfiniteScrollInternalProps<any>, IInfiniteScrollProps<any>>(
```

Comment:
Please add brief TypeDoc documentation to the InfiniteScroll component

Reply:
added